sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestTimestamped.java

```diff
@@ -31,94 +38,122 @@
         personDAO.createTable();
     }
 
     @Test
     public void shouldInsertCreatedAndModifiedFields() {
-        // This is one way we can get the binding information of the executed query
-        dbRule.getJdbi().setTimingCollector((elapsed, ctx) ->
-                assertThat(ctx.getBinding().findForName("now", ctx)).isPresent());
-
         Person p = new Person("John", "Phiri");
         p.setId(1);
-        personDAO.insert(p);
 
-        // Clear the timing colletor
-        dbRule.getJdbi().setTimingCollector(TimingCollector.NOP_TIMING_COLLECTOR);
-
-        Person found = personDAO.get(1);
+        logNextTimestamp("now");
+        personDAO.insert(p);
+        verifyLastTimestamp(T0);
 
-        assertThat(found.getCreated()).isNotNull();
-        assertThat(found.getModified()).isNotNull();
+        Person inserted = personDAO.get(1);
+        assertThat(inserted.getCreated()).isEqualTo(T0_TIMESTAMP);
+        assertThat(inserted.getModified()).isEqualTo(T0_TIMESTAMP);
     }
 
     @Test
     public void shouldAllowCustomTimestampParameter() {
-        LocalDateTime timeBefore = LocalDateTime.now().minusMinutes(1);
-
         Person p = new Person("John", "Phiri");
         p.setId(1);
-        dbRule.getJdbi().setTimingCollector((elapsed, ctx) ->
-                assertThat(ctx.getBinding().findForName("createdAt", ctx)).isPresent());
 
+        logNextTimestamp("createdAt");
         personDAO.insertWithCustomTimestampFields(p);
+        verifyLastTimestamp(T0);
 
-        dbRule.getJdbi().setTimingCollector(TimingCollector.NOP_TIMING_COLLECTOR);
-
-        Person fetched = personDAO.get(1);
-
-        assertThat(p.getFirstName()).isEqualTo(fetched.getFirstName());
-        assertThat(p.getLastName()).isEqualTo(fetched.getLastName());
-        assertThat(fetched.getCreated()).isNotNull();
-        assertThat(fetched.getModified()).isNotNull();
-
-        assertThat(fetched.getCreated()).isEqualTo(fetched.getModified());
-
-        assertThat(timeBefore).isBefore(fetched.getCreated().toLocalDateTime());
+        Person inserted = personDAO.get(1);
+        assertThat(p.getFirstName()).isEqualTo(inserted.getFirstName());
+        assertThat(p.getLastName()).isEqualTo(inserted.getLastName());
+        assertThat(inserted.getCreated()).isEqualTo(T0_TIMESTAMP);
+        assertThat(inserted.getModified()).isEqualTo(T0_TIMESTAMP);
     }
 
     @Test
     public void shouldUpdateModifiedTimestamp() {
-        Person p = new Person("John", "Phiri");
 
+        Person p = new Person("John", "Phiri");
         p.setId(3);
 
-        dbRule.getJdbi().setTimingCollector((elapsed, ctx) ->
-                assertThat(ctx.getBinding().findForName("now", ctx)).isPresent());
-
+        logNextTimestamp("now");
         personDAO.insert(p);
+        verifyLastTimestamp(T0);
 
-        dbRule.getJdbi().setTimingCollector(TimingCollector.NOP_TIMING_COLLECTOR);
+        when(clock.instant()).thenReturn(T0.plusSeconds(10).toInstant());
```

Comment:
In previous projects I've seen folks implement a `FakeClock` class, with methods like `advance(Duration)` to simplify simulating passage of time.